### PR TITLE
[Refacto] Migration due to tensordict 473 and 474

### PR DESCRIPTION
**Pull request by vmoens**

### CODE REVIEW

File: torchrl/modules/tensordict_module/common.py [hunk 1]
```diff
@@ -224,13 +225,17 @@ def register_spec(self, safe, spec):
         elif spec is None:
             spec = CompositeSpec()
 
-        if set(spec.keys(True, True)) != set(self.out_keys):
+        if sorted(unravel_key_list(list(spec.keys(True, True))), key=str) != sorted(
+            self.out_keys, key=str
```
Comment (matteobettini) [inline torchrl/modules/tensordict_module/common.py:229]: you might want to unravel here too

File: torchrl/modules/tensordict_module/common.py [hunk 2]
```diff
@@ -217,22 +218,25 @@ def register_spec(self, safe, spec):
...
             if "_" in spec.keys() and spec["_"] is not None:
                 warnings.warn('got a spec with key "_": it will be ignored')
         elif spec is None:
             spec = CompositeSpec()
 
-        if set(spec.keys(True, True)) != set(self.out_keys):
+        # unravel_key_list(self.out_keys) can be removed once 473 is merged in tensordict
+        spec_keys = sorted(unravel_key_list(list(spec.keys(True, True))), key=str)
+        out_keys = sorted(unravel_key_list(self.out_keys), key=str)
+        if set(spec_keys) != set(out_keys):
```
Comment (matteobettini) [inline torchrl/modules/tensordict_module/common.py:231]: we can anticipate the call to set() in line 229 and 230, like u do in probabilistic.py

**Review by matteobettini — APPROVED** [review]
LGTM

File: torchrl/modules/tensordict_module/common.py (hunk 2 above)
Comment (vmoens) [reply torchrl/modules/tensordict_module/common.py:231]: what do you mean exactly? Trying to solve it

File: torchrl/modules/tensordict_module/probabilistic.py [hunk 3]
```diff
@@ -129,22 +129,24 @@ def __init__(
...
+            spec = CompositeSpec({self.out_keys[0]: spec})
         elif spec is not None and isinstance(spec, CompositeSpec):
             if "_" in spec.keys():
                 warnings.warn('got a spec with key "_": it will be ignored')
         elif spec is None:
             spec = CompositeSpec()
-
-        if set(spec.keys(True, True)) != set(self.out_keys):
+        spec_keys = set(unravel_key_list(list(spec.keys(True, True))))
+        out_keys = set(unravel_key_list(self.out_keys))
```
Comment (matteobettini) [inline torchrl/modules/tensordict_module/probabilistic.py:139]: like here

**Pull request merged**